### DISCUSSION TITLE
Only detect people about to ring

### DISCUSSION THREAD

**Discussion by kcirtaph** [blakeblackshear/frigate#3515] [Ask A Question] [answered]
Also a Frigate fan for a couple of weeks now!

My door and doorbird camera are very close to the street. Therefore it picks up a lot of people who pass by at 2m (7 ft) or so. 

My aim is to only detect people who are about to ring or are hanging around. Is there a way that I can only get a notification in HA if 
- people are standing for X amount of time in the feed and/or 
- are very close to the camera (take up certain share of the feed)?

**Comment by ediguidedog** [comment]
There may be better ways but assuming you can't set a zone that only covers people about to ring your doorbell you could have a look at the min_area: setting objects: > filter: > person: >  min_area:  . To see the size of any object turn on the bounding boxes on the debug view. The number next to the percentage is the count of pixels that is tested against the min_area setting 

![image](https://user-images.githubusercontent.com/73383390/180642778-f069ff32-d1df-4e7f-b1dd-a5f290f46fdd.png)

**Reply by kcirtaph** [reply]
awesome that's very helpful. unfortunately indeed I cannot carve out an area for people about to ring. 

there is no minimum duration setting similar to minimum pixels by any chance?

**Reply by NickM-27** [reply]
As far as notifications go you can use the MQTT events topic to filter this out. There's not a good way besides zones to do this to limit events however

**Comment by blakeblackshear** [comment, marked as answer]
I wouldn't use the min_area filter here. That's really meant for false positives. I would use the attributes in the events mqtt topic. There are several attributes there you can use to determine a minimum size and where the person is standing.
